### ISSUE TITLE
Add a new way of shading meshes, better than flat or, smooth shading . . .

### ISSUE THREAD

**Issue by jasperbrooks79** [godotengine/godot-proposals#1682]
**Describe the project you are working on:**
A Tomb raider -adventure game . .

**Describe the problem or limitation you are having in your project:**
I've been looking at a lot of titles, that have smooth shading ( PS3, mostly ), and compared it, to flat shading, and I've been wondering, we've had smooth shading, for a long time, but are there ' improved ' versions, of these two standards ways, of shading . . I know normal maps are, like the next step, but I was wondering, if there was a ' level ' in between, where one could could get a ' nicer ' version, of smooth shading . . Been looking for it, on the internet, found this . .
![Skærmbillede (559)](https://user-images.githubusercontent.com/61069740/96360814-bba96a80-1120-11eb-812a-ff648aaeb95d.jpg)

I'm sort of asking, if there is a way, to take ' smooth ' shading, to the next level, with a setting one could set, under shading technique, in materials, or so . . we've had smooth shading for a LONG time, and was wondering if there have been some improvements in that rendering style, that might be fast, require no normal maps, and give better, more realistic results, compared to either smooth, or flat shading . .
![2020-10-18 0904](https://user-images.githubusercontent.com/61069740/96360836-ed223600-1120-11eb-9ce6-7fc66deef8dd.jpg)

I'm not an expert, of 3d rendering, but maybe there is a way, to add an ' improved smooth ' shading option, that one could set, where one sets ' toon ' shading, or so, in Godot, and sort of, make better renders, of 3d meshes, that are in smooth shading, I don't know enough 3d for what to look for, but it would be really cool, imo, Thanks <3

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
A sort, of ' improved smooth ' shading technique, if it exists, I'm asking bec. I can't find it, but the engine pros MIGHT have heard of it, something that is fast, performance and, gives nice visual results, or so . .
![2020-10-18 0907](https://user-images.githubusercontent.com/61069740/96360885-586c0800-1121-11eb-9f23-91fe3dbea125.jpg)

Smooth shading gives ok, but also ' ugly-ish ' results, ie. one can see artifacts, so on . . I was wondering if smooth shading might be improved, so it looked even nicer, with some techniques, to get even better, more realistic results, in Godot . . Something fast, maybe shader based, not sure, or I can't imagine the old ' smooth ' shading technique hasn't been improved, already, by researchers, or so, but I've been looking for it, and can't find it, anyway wanted to ask, since the engine coders might know some really cool stuff, the rest of us never heard about <3 . . . Thanks . .

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Well, take the old technique, of smooth shading, and improve it, maybe by generating in-between vertices, or some fancy new method, of getting even nicer renders, of surfaces, with no normal maps, etc etc . . there might be a cheap way, of making every-thing look nicer, right away, no baking, no normal maps, etc . . it wouldn't replace normal maps, but sort of be a step towards normal maps, the way smooth shading was, for flat shading, ie. ' improved / better smooth shading ' technique . . One would assume, there has been improvements to this shading technique, since it's been around, for many years, and is heavily used, some-thing fast, or quick, that improves quality . .

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I can't rewrite basic stuff, in the engine, still learning, to make code, for games . .

**Is there a reason why this should be core and not an add-on in the asset library?:**
Well, if such a method exists, might be ' just ' the look one wants, and lessen need, for normal maps, or so . .

Basically, it'd be something, that if one imports a ' smooth ' shading object, it just looks better, no need for normal maps, would be nice to have that, again, not shader expert, might, be dumb . . <3 some-thing, to bridge the gap, between the ' toon ' looking smooth shading, see sec. image above, and full normal maps, a 'nicer' smooth shading method or, algorhitm . . <3 Thx . . .

**Comment by jasperbrooks79** [comment]
I know this is vague, maybe there are some super - coders here, who know, what it could be or, could develop it, or so . . <3 not, sure . . . <3

**Comment by jasperbrooks79** [comment]
ps. I haven't been involved in 3d rendering, for the last many years, so I have got no idea, what it could be called, or so . . Maybe there's a 3d - god here, who knows really hard-core stuff, that has the answer, so we can get something ' between ' smooth shading, and normal maps, or so . .

**Comment by Calinou** [comment]
Weighted face normals can be created in your 3D DCC of choice (Blender has a modifier for it since 2.80). Modifying normals at import-time is definitely out of scope for a game engine like Godot.

**Comment by Calinou** [comment]
Closing, as weighted face normals should be created in 3D modeling software as mentioned above. When performing procedural geometry generation, you can create weighted face normals by customizing the normal direction (and not relying on automatic normal generation).
